### PR TITLE
Make sure the Glean test looks for the right 'reason'

### PR DESCRIPTION
**Pull request by Dexterp37**
---

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by Dexterp37** [comment]
bors try

**Comment by Dexterp37** [comment]
Looks like this is [passing tests on CI now](https://firefox-ci-tc.services.mozilla.com/tasks/HueK1GaOReKzhZiPFXUEIg), while it was consistently failing before.

File: samples/glean/src/androidTest/java/org/mozilla/samples/glean/pings/BaselinePingTest.kt [hunk 1]
```diff
@@ -63,7 +64,16 @@ class BaselinePingTest {
             val request = server.takeRequest(20L, TimeUnit.SECONDS)
             val docType = request.path.split("/")[3]
             if (pingName == docType) {
-                return JSONObject(request.body.readUtf8())
+                val parsedPayload = JSONObject(request.body.readUtf8())
+                if (pingReason == null) {
+                    return pingReason
```
Comment (mdboom) [inline samples/glean/src/androidTest/java/org/mozilla/samples/glean/pings/BaselinePingTest.kt:69]: Should this return parsedPayload here?

Comment (Dexterp37) [reply samples/glean/src/androidTest/java/org/mozilla/samples/glean/pings/BaselinePingTest.kt]: Yes, good catch

**Comment by Dexterp37** [comment]
bors try

**Comment by Dexterp37** [comment]
Ok, the [related task is green again](https://firefox-ci-tc.services.mozilla.com/tasks/VSz0qb3STnCzFUcjMNRXRw). Merging this!

**Pull request merged**